Classes/Composer/ComposerPackageManager.php

```diff
@@ -0,0 +1,257 @@
+                continue;
+            }
+            $this->addPackageInfo($packageInfo);
+        }
+    }
+
+    /**
+     * Process all composer installed packages.
+     */
+    private function processPackages(): void
+    {
+        foreach (InstalledVersions::getAllRawData() as $loader) {
+            foreach ($loader['versions'] as $packageName => $version) {
+                $packagePath = $this->getPackageInstallPath($packageName);
+                $packageRealPath = $this->realPath($packagePath);
+                $info = $this->getPackageComposerJson($packagePath) ?? [];
+                $packageType = $info['type'] ?? '';
+                $this->addPackageInfo(new PackageInfo(
+                    $packageName,
+                    $packageType,
+                    $packagePath,
+                    $packageRealPath,
+                    (string)($version['pretty_version'] ?? ''),
+                    $info
+                ));
+            }
+        }
+    }
+
+    /**
+     * Adds the package information to the internal cache. Additionally, it sets the extensionKey to packageName
+     * map information, if a TYPO3 extension or system-extensions package information is handed over. This map
+     * is used to allow extensionKey or packageName for retrieving package information, which comes in handy to
+     * provide backward compatibility for test core- and test extension symlink configuration per test instance.
+     */
+    private function addPackageInfo(PackageInfo $packageInfo): void
+    {
+        if (self::$packages[$packageInfo->getName()] ?? null) {
+            return;
+        }
+        self::$packages[$packageInfo->getName()] = $packageInfo;
+        if ($packageInfo->getExtensionKey() !== '') {
+            self::$extensionKeyToPackageNameMap[$packageInfo->getExtensionKey()] = $packageInfo->getName();
+        }
+    }
+
+    private function rootPackage(): ?PackageInfo
+    {
+        return self::$rootPackage;
+    }
+
+    private function getPackageComposerJson(string $path): ?array
+    {
+        $composerFile = rtrim($path, '/') . '/composer.json';
+        if (!file_exists($composerFile) || !is_readable($composerFile)) {
+            return null;
+        }
+        try {
+            return json_decode((string)file_get_contents($composerFile), true, JSON_THROW_ON_ERROR);
+        } catch(\Throwable) {
+            // skipped
+        }
+        return null;
+    }
+
+    private function resolvePackageName(string $name): string
+    {
+        if (str_starts_with($name, 'typo3conf/ext/')
+            || str_starts_with($name, 'typo3/sysext/')
+        ) {
+            $name = basename($name);
+        }
+        return self::$extensionKeyToPackageNameMap[$name] ?? $name;
+    }
+
+    /**
+     * Get the sanitized package installation path.
+     *
+     * Note: Not using realpath() is done by intention. That gives us the ability, to eventually avoid duplicates and
```

Comment:
Should this be "eventually" (i.e., in the end) or "potentially" (i.e., possibly)?